### PR TITLE
Feat/rework cli

### PR DESCRIPTION
**Pull request by LuisBlanche**
Allows Initiation of project by adding methods to the python client for the API  and a user friendly CLI

### CODE REVIEW

**Comment by LuisBlanche** [comment]
I just need to figure out what is wrong with one of the tests  before submitting fo review ?

**Comment by benoit-cty** [comment]
> I just need to figure out what is wrong with one of the tests before submitting fo review ?

The return of `codecarbon config --init` is "1" (see [here](https://github.com/mlco2/codecarbon/actions/runs/8390694308/job/22979390317?pr=506#step:6:877)) so maybe try to add the texte output of the command to find out.

Does it works when you run `tox` locally ?

**Comment by LuisBlanche** [comment]
> > I just need to figure out what is wrong with one of the tests before submitting fo review ?
> 
> The return of `codecarbon config --init` is "1" (see [here](https://github.com/mlco2/codecarbon/actions/runs/8390694308/job/22979390317?pr=506#step:6:877)) so maybe try to add the texte output of the command to find out.
> 
> Does it works when you run `tox` locally ?

It's all good I just had a test that was creating a `.codecarbon.config` dile that was messing with the other tests. It's good for review now

**Comment by LuisBlanche** [comment]
I changed the base branch back and forth because the PR had not caught the rebase somehow. It's all good now

**Comment by LuisBlanche** [comment]
Here is a video made with asciinema of how the CLI works :
[https://asciinema.org/a/648518](https://asciinema.org/a/648518)

There is probably a way to integrate this video in the docs .
There is a special player that you can embed and you can actually copy and paste from the video

File: docs/edit/usage.rst [hunk 1]
```diff
@@ -16,12 +16,28 @@ Command line
...
+
+Start monitoring the emissions of the computer
+.. code-block:: console
+
+    codecarbon monitor  
 
 You have to stop the monitoring manually with ``Ctrl+C``.
 
+In the following example you will see how to use the CLI to monitor all the emissions of you computer and sending everything 
+to an API running on localhost:8008 (that you can start with the docke-compose)
```
Comment (benoit-cty) [inline docs/edit/usage.rst:33]:
```suggestion
to an API running on "https://api.codecarbon.io" (Or you can start a private local API with "docke-compose up")
```

**Review by benoit-cty — COMMENTED** [review]
When I run `codecarbon config` I just get an empty line.

My existing config:
```
❯ cat .codecarbon.config 
[codecarbon]
experiment_id = 3b1fd3cb-488a-4bcf-b727-ff96dfd782db
log_level = DEBUG
```

**Review by LuisBlanche — COMMENTED** [review]
I have rebased from master to adapt to API changes and made some fixes and upgrades , I've made new videos here : 

https://asciinema.org/~LuisBlanche

File: docs/usage.html [hunk 2]
```diff
@@ -122,10 +120,23 @@ <h2>Online Mode<a class="headerlink" href="#online-mode" title="Link to this hea
...
+<div><p>codecarbon config –init</p>
+</div></blockquote>
+<p>Start monitoring the emissions of the computer
+.. code-block:: console</p>
+<blockquote>
+<div><p>codecarbon monitor</p>
+</div></blockquote>
 <p>You have to stop the monitoring manually with <code class="docutils literal notranslate"><span class="pre">Ctrl+C</span></code>.</p>
+<p>In the following example you will see how to use the CLI to monitor all the emissions of you computer and sending everything
+to an API running on localhost:8008 (that you can start with the docke-compose)
```
Comment (inimaz) [inline docs/usage.html:135]: `docker-compose`. 

We could add a `Coming soon` comment or something mentioning that this only works in a local instance of the server for now.

**Review by inimaz — APPROVED** [review]
Looks good to me! Nice @LuisBlanche ! Thanks for this. I have left a small comment

**Pull request merged**